documentation/src/main/asciidoc/cheat-sheets/cache-configuration.asciidoc

```diff
@@ -0,0 +1,435 @@
+= {brandname} cache configuration cheat sheet
+
+Use the following information and configuration examples as a quick reference for creating and configuring {brandname} caches.
+
+== Clustered cache modes
+
+Distributed:: Distributed caches offer linear scalability and efficient memory usage by creating fewer copies of each entry across the cluster.
+Distributed caches store a number of copies for each cache entry that is equal to the *Number of owners* that you configure.
++
+[source,xml]
+----
+<distributed-cache />
+----
+
+Replicated:: Replicated caches offer data guarantees and fault tolerance by creating a copy of all entries on each node in the cluster.
++
+[source,xml]
+----
+<replicated-cache />
+----
+
+== Cluster replication
+
+Synchronous replication:: Helps to keep your data consistent but adds latency to cluster traffic that reduces throughput for cache writes.
++
+[source,xml]
+----
+<distributed-cache mode="SYNC" />
+----
+Asynchronous replication:: Reduces latency and increases the speed of write operations but leads to data inconsistency and provides a lower guarantee against data loss.
++
+[source,xml]
+----
+<distributed-cache mode="ASYNC" />
+----
+
+== Cache configuration options
+
+.Encoding
+A format, identified by a media type, that {brandname} uses to store keys and values.
+
+TIP: {brandname} recommends the `application/x-protostream` media type. Protobuf gives you excellent performance as well as interoperability between client applications in different programming languages for both Hot Rod and REST endpoints.
```

Comment:
Note that the marshaller must work with that encoding. Otherwise there are errors if a client push entries with the wrong marshaller.
I see user issues with that wrong configuration.
I.e. JavaSerialization or JBossMarshalling are not working with protostream encoding